editor/tree_search.h

```diff
@@ -0,0 +1,126 @@
+/**
+ * tree_search.h
+ * =============================================================================
+ * Copyright 2021-2024 Serhii Snitsaruk
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ * =============================================================================
+ */
+
+#ifdef TOOLS_ENABLED
+
+#ifndef TREE_SEARCH_H
+#define TREE_SEARCH_H
+
+#ifdef LIMBOAI_GDEXTENSION
+#include <godot_cpp/classes/check_box.hpp>
+#include <godot_cpp/classes/h_flow_container.hpp>
+#include <godot_cpp/classes/label.hpp>
+#include <godot_cpp/classes/line_edit.hpp>
+#include <godot_cpp/classes/tree.hpp>
+#include <godot_cpp/templates/hash_map.hpp>
+#endif // LIMBOAI_GDEXTENSION
+
+#ifdef LIMBOAI_MODULE
+#include "scene/gui/check_box.h"
+#include "scene/gui/flow_container.h"
+#include "scene/gui/label.h"
+#include "scene/gui/line_edit.h"
+#include "scene/gui/tree.h"
+#include "core/templates/hash_map.h"
+#endif // LIMBOAI_MODULE
+
+using namespace godot;
+
+enum TreeSearchMode {
+	HIGHLIGHT = 0,
+	FILTER = 1
+};
+
+class TreeSearchPanel : public HFlowContainer {
```

Comment:
No! We had a HBoxContainer before [f523af2](https://github.com/limbonaut/limboai/pull/229/commits/f523af2d13bea14e09ba8dcae92c09ae5f67505c).
HFlowContainer allows for smaller minimum sizes. Otherwise we'd need something like a ScrollContainer in-between. I tried that. Did increase minimum v-size. Didn't like that. What do you think?

Reply:
HBox's minimum is the combined minimum of its children, and the smallest size it can have is zero size. I don't think I understand what you mean. I switched it to HBoxContainer - and took a screenshot - identical result. Am I missing something?

Reply:
current - `HFlowContainer`
[demo_hflowcontainer.webm](https://github.com/user-attachments/assets/f032176b-5619-4865-a8cc-3ee89151d4b1)

[a80b737](https://github.com/limbonaut/limboai/pull/229/commits/a80b737319da1ffeead4c3e250f303d6844817ae) `HBoxContainer`
[demo_hboxcontainer.webm](https://github.com/user-attachments/assets/2f588f16-4b50-4970-92af-c3ec70131dda)

Reply:
Oh wow, okay, that's super trippy. If it works better with HFlow, let's keep that, then.